src/shared/tests.h

```diff
@@ -200,3 +200,42 @@ static inline int run_test_table(void) {
         DEFINE_TEST_MAIN_FULL(log_level, intro, NULL)
 #define DEFINE_TEST_MAIN(log_level)                     \
         DEFINE_TEST_MAIN_FULL(log_level, NULL, NULL)
+
+#define ASSERT_OK(expr)                                                                                  \
+        ({                                                                                               \
+                int _result = (expr);                                                                    \
+                if (_result < 0) {                                                                       \
+                        log_error_errno("Assertion failed: %s (result: %d, error: %m)", #expr, _result); \
+                        abort();                                                                         \
+                }                                                                                        \
+        })
+
+#define ASSERT_EQ(expr1, expr2)                                                                          \
+        ({                                                                                               \
+                int _expr1 = (expr1);                                                                    \
+                int _expr2 = (expr2);                                                                    \
```

Comment:
Please use `typeof(expr1)` instead of `int`

Reply:
thanks for this

`
#define ASSERT_EQ(expr1, expr2)                                                                          \
        ({                                                                                               \
                typeof(expr1) _expr1 = (expr1);                                                                    \
                typeof(expr1) _expr2 = (expr2);                                                                    \
                if (_expr1 != _expr2) {                                                                  \
                        log_error("Assertion failed: expected %s == %s, but %lf != %lf", #expr1, #expr2, _expr1, _expr2); \
                        abort();                                                                         \
                }                                                                                        \
        })
`

it will result into this right ?

Reply:
@DaanDeMeyer since we are logging it using %d, is there a need to use typeof ?, I also try using the %lf hoping to typecast, but it was giving warning. Thank you.